src/jit/importer.cpp

```diff
@@ -4390,6 +4394,311 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
             }
             break;
         }
+
+        case NI_Base_Vector256_WithElement:
+        {
+            if (!compSupports(InstructionSet_AVX))
+            {
+                // Using software fallback if JIT/hardware don't support AVX instructions and YMM registers
+                return nullptr;
+            }
+            __fallthrough;
+        }
+        case NI_Base_Vector128_WithElement:
+        {
+            assert(sig->numArgs == 2);
+            GenTree* indexOp = impStackTop(1).val;
+            if (!compSupports(InstructionSet_SSE2) || !varTypeIsArithmetic(baseType) || !indexOp->OperIsConst())
+            {
+                // Using software fallback if
+                // 1. JIT/hardware don't support SSE2 instructions
+                // 2. baseType is not a numeric type (throw execptions)
+                // 3. index is not a constant
+                return nullptr;
+            }
+
+            switch (baseType)
+            {
+                // Using software fallback if baseType is not supported by hardware
+                case TYP_BYTE:
+                case TYP_UBYTE:
+                case TYP_INT:
+                case TYP_UINT:
+                case TYP_FLOAT:
+                    if (!compSupports(InstructionSet_SSE41))
```

Comment:
There are other libraries (such as DirectXMath) that do this for "downlevel" hardware via a:
* `Shuffle` (swap 0 and index being inserted)
* `Replace` (set element 0 to the inserted value)
* `Shuffle` (swap 0 and the index being inserted)

Reply:
Ah, yes, but I think two `shufps` are enough to emulate. Will try tonight.